src/index.ts

```diff
@@ -1,53 +1,92 @@
-import { AnyLayer, IControl, Map } from 'maplibre-gl';
+import {ControlPosition, IControl, LayerSpecification, Map} from 'maplibre-gl';
 
-import { play, pause, reload, skipBackward, skipForward } from './icons';
+import {pause, play, reload, skipBackward, skipForward} from './icons';
 
 const ACTIVE_BUTTON_COLOR = 'rgb(204, 204, 204)';
 
 type ContainerOptions = {
-    length: number;
-    interval: number;
-    onSliderValueChange: () => void;
+  length: number;
+  interval: number;
+  loopDelay: number;
+  showButtons?: boolean;
+  onSliderValueChange: () => void;
+  loop?: boolean;
+  autoplay?: boolean;
 };
 
 const makeImg = (icon: string): HTMLImageElement => {
-    const img = document.createElement('img');
-    img.src = icon;
-    return img;
+  const img = document.createElement('img');
+  img.src = icon;
+  return img;
 };
 
 const makeContainer = ({
-    length,
-    interval,
-    onSliderValueChange,
-}: ContainerOptions): [HTMLDivElement, HTMLDivElement, HTMLInputElement] => {
-    // outest div
-    const container = document.createElement('div');
-    container.classList.add('mapboxgl-ctrl');
-    container.classList.add('mapboxgl-ctrl-group');
-    container.style.width = '240px';
-    container.style.height = '84px';
-    container.style.backgroundColor = '#fff';
-    container.style.textAlign = 'center';
-
-    const titleDiv = document.createElement('div');
-    titleDiv.innerHTML = '<br />';
-    titleDiv.style.marginTop = '4px';
-    container.appendChild(titleDiv);
-
-    // temporal slider
-    const slider = document.createElement('input');
-    slider.type = 'range';
-    slider.value = '0';
-    slider.min = '0';
-    slider.max = String(length - 1);
-    slider.addEventListener('input', () => {
+                         length,
+                         interval,
+                         showButtons,
+                         onSliderValueChange,
+                         loop,
+                         loopDelay,
+                         autoplay
+                       }: ContainerOptions): [HTMLDivElement, HTMLDivElement, HTMLInputElement] => {
+  let looping = loop || false;
+  let playing = autoplay || false;
+  let playingTimer: any = null;
```

Comment:
- Is it better to use same words to same means? (`loop`)
- I want to add type for `playingTimer`. is following typing correct?

```suggestion
  let _loop = loop;
  let playing = autoplay;
  let playingTimer: NodeJS.Timeout | null = null;
```